examples/model_selection/plot_grid_search_stats.py

```diff
@@ -0,0 +1,506 @@
+#
+# We can carry out our pairwise comparisons the same way as in the first
+# section:
+
+pairwise_bayesian = []
+
+for model_i, model_k in combinations(range(len(model_scores)), 2):
+    model_i_scores = model_scores.iloc[model_i].values
+    model_k_scores = model_scores.iloc[model_k].values
+    differences = model_i_scores - model_k_scores
+    t_post = t(
+        df, loc=np.mean(differences),
+        scale=correct_std(differences, n, n_train, n_test)
+        )
+    left_prob = t_post.cdf(rope_interval[0])
+    right_prob = t_post.sf(rope_interval[1])
+    rope_prob = t_post.cdf(rope_interval[1]) - t_post.cdf(rope_interval[0])
+
+    pairwise_bayesian.append([left_prob, right_prob, rope_prob])
+
+pairwise_bayesian_df = (pd.DataFrame(
+    pairwise_bayesian,
+    columns=['worse_prob', 'better_prob', 'rope_prob']
+    ).round(3))
+
+pairwise_comp_df = pairwise_comp_df.join(pairwise_bayesian_df)
+pairwise_comp_df
+
+
+###############################################################################
+# Using the Bayesian approach we can compute the probability that a model
+# performs better, worse or practically equivalent to another.
+#
+# Results show that the model ranked first by our GridSearch, `rbf`, has
+# approximately a 6.8% chance of being worse than `linear`, and a 1.8% chance
+# of being worse than `poly3`.
+#
+# `rbf` and `linear` have a 43% probability of being practically equivalent,
+# while `rbf` and `poly3` have a 10% chance of being so.
+#
+# Similarly to the conclusions obtained using the Frequentist approach, all
+# models have a 100% probability of being better than `poly2`, and none have a
+# practically equivalent performance with the latter.
+
+
+###############################################################################
+# Take-home messages
+# ------------------
+# - When statistically comparing the performance of two models evaluated in
+#   GridSearchCV, it is necessary to correct the calculated variance that could
+#   be underestimated given that the scores of the models are not independent
+#   from each other.
+# - A Frequentist approach that uses a (variance-corrected) paired t-test can
+#   tell us if the performance of one model is better that another with a
+#   degree of certainty above chance.
+# - A Bayesian approach can provide the probabilities of one model being
+#   better, worse or practically equivalent than another. It can also tell us
+#   how confident we are of knowing that the true differences of our models
+#   fall under a certain range of values.
+# - If multiple models are statistically compared, a multiple comparisons
+#   correction is needed when using the Frequentist approach.
+
+
+###############################################################################
+# References:
+# ___________
+# .. [1] Dietterich, T. G. (1998). Approximate statistical tests for comparing
+#        supervised classification learning algorithms. Neural computation,
+#        10(7).
```

Comment:
Adding a link to the paper (non-paywall if possible)  would also be great, e.g.,:

```
# .. [1] Dietterich, T. G. (1998). `Approximate statistical tests for comparing
#        supervised classification learning algorithms
#        <http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.37.3325&rep=rep1&type=pdf>`_.
#        Neural computation, 10(7).
```